File: app/src/main/java/pl/srw/billcalculator/bill/calculation/PgnigCalculatedBill.java
package pl.srw.billcalculator.bill.calculation;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.math.RoundingMode;

import lombok.Getter;
import pl.srw.billcalculator.pojo.IPgnigPrices;

/**
 * Created by Kamil Seweryn.
 */
@SuppressWarnings("FieldCanBeLocal")
@Getter
public class PgnigCalculatedBill extends CalculatedBill {

    private final int consumptionM3;
    private final BigInteger consumptionKWh;

    private final BigDecimal oplataAbonamentowaNetCharge;
    private final BigDecimal paliwoGazoweNetCharge;
    private final BigDecimal dystrybucyjnaStalaNetCharge;
    private final BigDecimal dystrybucyjnaZmiennaNetCharge;

    private final BigDecimal oplataAbonamentowaVatCharge;
    private final BigDecimal paliwoGazoweVatCharge;
    private final BigDecimal dystrybucyjnaStalaVatCharge;
    private final BigDecimal dystrybucyjnaZmiennaVatCharge;

    public PgnigCalculatedBill(final int readingFrom, final int readingTo, final String dateFrom, final String dateTo, final IPgnigPrices prices) {
        super(true, dateFrom, dateTo);
        consumptionM3 = readingTo - readingFrom;
        consumptionKWh = new BigDecimal(consumptionM3).multiply(new BigDecimal(prices.getWspolczynnikKonwersji()))
                .setScale(0, RoundingMode.HALF_UP).toBigInteger();

        oplataAbonamentowaNetCharge = countNetAndAddToSum(prices.getOplataAbonamentowa(), getMonthCount());
        paliwoGazoweNetCharge = countNetAndAddToSum(prices.getPaliwoGazowe(), consumptionKWh);
        dystrybucyjnaStalaNetCharge = countNetAndAddToSum(prices.getDystrybucyjnaStala(), getMonthCount());
        dystrybucyjnaZmiennaNetCharge = countNetAndAddToSum(prices.getDystrybucyjnaZmienna(), consumptionKWh);

        oplataAbonamentowaVatCharge = countVatAndAddToSum(oplataAbonamentowaNetCharge);
        paliwoGazoweVatCharge = countVatAndAddToSum(paliwoGazoweNetCharge);
        dystrybucyjnaStalaVatCharge = countVatAndAddToSum(dystrybucyjnaStalaNetCharge);
        dystrybucyjnaZmiennaVatCharge = countVatAndAddToSum(dystrybucyjnaZmiennaNetCharge);

    }
}

File: app/src/main/java/pl/srw/billcalculator/bill/calculation/PgnigCalculatedBill.kt
package pl.srw.billcalculator.bill.calculation

import org.threeten.bp.LocalDate
import pl.srw.billcalculator.pojo.IPgnigPrices
import pl.srw.billcalculator.util.Dates
import java.math.BigDecimal
import java.math.RoundingMode

class PgnigCalculatedBill(readingFrom: Int, readingTo: Int, dateFrom: LocalDate, dateTo: LocalDate, prices: IPgnigPrices)
    : CalculatedBill(true, dateFrom, dateTo) {

    val monthCountExact = Dates.countMonth(dateFrom, dateTo)

    val consumptionM3 = readingTo - readingFrom
    val consumptionKWh = BigDecimal(consumptionM3).multiply(BigDecimal(prices.wspolczynnikKonwersji))
            .setScale(0, RoundingMode.HALF_UP).toBigInteger()

    val oplataAbonamentowaNetCharge = countNetAndAddToSum(prices.oplataAbonamentowa, monthCount)
    val paliwoGazoweNetCharge = countNetAndAddToSum(prices.paliwoGazowe, consumptionKWh)
    val dystrybucyjnaStalaNetCharge = countNetAndAddToSum(prices.dystrybucyjnaStala, monthCountExact)
    val dystrybucyjnaZmiennaNetCharge = countNetAndAddToSum(prices.dystrybucyjnaZmienna, consumptionKWh)

    val oplataAbonamentowaVatCharge = countVatAndAddToSum(oplataAbonamentowaNetCharge)
    val paliwoGazoweVatCharge = countVatAndAddToSum(paliwoGazoweNetCharge)
    val dystrybucyjnaStalaVatCharge = countVatAndAddToSum(dystrybucyjnaStalaNetCharge)
    val dystrybucyjnaZmiennaVatCharge = countVatAndAddToSum(dystrybucyjnaZmiennaNetCharge)
}